### PR TITLE
Remove 'Outside of World' message in GuiOverlayDebug

### LINKED ISSUE #54: Debug screen shows "Outside of world..." when above y=255/below y=0

### PR DESCRIPTION
**Pull request by Boreeas**
Fixes #54

### CODE REVIEW

File: src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	private int blockPosGetYRedirect(BlockPos pos) {// In bounds - don't modify anything
+		if (pos.getY() >= 0 && pos.getY() < 256) return pos.getY();
+
+		// Retain normal semantics for non-cubic worlds
+		if (mc.theWorld instanceof ICubicWorldClient) {
+			return MathHelper.clamp_int(pos.getY(), 0, 255);
+		}
+
+		return pos.getY();
```
Comment (Barteks2x) [inline src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java]: There is already Util class with method that does exactly that but better and only for CubicChunks world type. Look at `MixinUtils` class.

File: src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import static cubicchunks.asm.JvmNames.BLOCK_POS_GETY;
+
+/**
+ * Goal of this mixin is to remove the "Outside of world..." message on the debug overlay for cubic world types.
+ * We redirect the call to BlockPos.getY() for the bounds check in GuiOverlayDebug.call only
+ * @author Malte Schütze
+ */
+@Mixin(GuiOverlayDebug.class)
+public class MixinGuiOverlayDebug {
+	@Shadow private Minecraft mc;
```
Comment (Barteks2x) [inline src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java]: Add `@Final`. Not a big change, but Mixin will complain that it's not there.

File: src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	private int blockPosGetYBoundsCheck2(BlockPos pos) {
+		return blockPosGetYRedirect(pos);
+
+	}
+
+	private int blockPosGetYRedirect(BlockPos pos) {// In bounds - don't modify anything
+		if (pos.getY() >= 0 && pos.getY() < 256) return pos.getY();
+
+		// Retain normal semantics for non-cubic worlds
+		if (mc.theWorld instanceof ICubicWorldClient) {
```
Comment (Barteks2x) [inline src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java]: All worlds are instances of ICubicWorld(Client). It's `isCubicWorld` method that says if it's CubicChunks worldd. See comment below.

File: src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java (hunk 1 above)
Comment (Boreeas) [inline src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java]: I was looking at MixinUtils, but it seemed to do the opposite of what I wanted to do - return the true value for out-of-bounds Y values, and fixed 64 for inside the bounds

Comment (Barteks2x) [inline src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java]: So this is the code:
```
		if (y < world.getMinHeight() || y >= world.getMaxHeight()) {
			return y;
		}
		return 64;
```
If we are out of bounds it returns the true value. And if we are out of bounds of the CubicChunks world - we are _definitely_ out of bounds of vanilla world. If we are _not_ out of bounds of cubic chunks world - return some value that won't trigger the out of bounds check. 64 is fine.

Comment (Boreeas) [inline src/main/java/cubicchunks/asm/mixin/noncritical/client/MixinGuiOverlayDebug.java]: Ah, I was assuming that min/max height were the vanilla values, i.e. 0, 255.

**Pull request merged**